### ISSUE TITLE
Style/PercentLiteralDelimiters does dot detect wrong delimiters if string contains `(`

### ISSUE THREAD

**Issue by noniq** [bbatsov/rubocop#3403]
With default settings, each of the following lines should trigger the Style/PercentLiteralDelimiters cop (*“%-literals should be delimited by ( and ).”*), but none of them does:

```ruby
puts %{(}
puts %{ ) }
puts %[( )]
```

It seems to make no difference which delimiters are used: As long as the string contains at least a single opening or closing paren, the cop isn’t triggered.

--------

## Expected behavior
Style/PercentLiteralDelimiters cop triggered.

## Actual behavior
Style/PercentLiteralDelimiters cop not triggered.

## Steps to reproduce the problem
Run RuboCop with default settings on the Ruby code given above.

## RuboCop version

```
$ rubocop/bin/rubocop -V
0.42.0 (using Parser 2.3.1.2, running on ruby 2.1.6 x86_64-darwin14.0)
$ (cd rubocop && git rev-parse HEAD)
429153801d1922bd7f2cc8ab577374cea7017630
```

**Comment by Drenmi** [comment]
Looking at the unit tests, this behaviour is intentional. Most likely to allow you to still be able to have string literals with parentheses.

**Comment by noniq** [comment]
Thanks, makes totally sense. Maybe the message should be changed to something like “%-literals that do not contain parens should be delimited by ( and )” to avoid this kind of confusion in the future? Not sure if this is really necessary, though.
